rewrite-java/src/main/java/org/openrewrite/java/CreateJavaClass.java

```diff
@@ -0,0 +1,182 @@
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.CreateFileVisitor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.SourceFile;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class CreateJavaClass extends ScanningRecipe<AtomicBoolean> {
+
+    @JsonIgnore
+    @Language("java")
+    private static final String NEW_CLASS_TEMPLATE = "package %s;\n\npublic class %s {\n}";
+
+    @Option(displayName = "Source root",
+            description = "The source root of the new class file.",
+            valid = {"main", "test"},
+            example = "main")
+    @NonNull
+    String sourceRoot;
+
+    @Option(displayName = "Package name",
+            description = "The package of the new class.",
+            example = "org.openrewrite.example")
+    @NonNull
+    String packageName;
+
+    @Option(displayName = "Class name",
+            description = "File path of new file.",
+            example = "ExampleClass")
+    @NonNull
+    String className;
+
+    @Language("java")
+    @Option(displayName = "Class template",
+            description = "The class template to be used.",
+            example = "Some text.")
+    @NonNull
+    String classTemplate;
+
+    @Option(displayName = "Overwrite existing file",
+            description = "If there is an existing file, should it be overwritten.",
+            required = false)
+    @Nullable
+    Boolean overwriteExisting;
+
+    @Option(displayName = "Relative directory path",
+            description = "Directory path of new class.",
+            required = false,
+            example = "foo/bar")
+    @Nullable
+    String relativePath;
+
+    @JsonCreator
+    public CreateJavaClass(
+            @JsonProperty("sourceRoot") String sourceRoot,
+            @JsonProperty("packageName") String packageName,
+            @JsonProperty("className") String className,
+            @Language("java") @JsonProperty("classTemplate") String classTemplate,
+            @JsonProperty("overwriteExisting") @Nullable Boolean overwriteExisting,
+            @JsonProperty("relativePath") @Nullable String relativePath
+    ) {
+        this.sourceRoot = sourceRoot;
+        this.packageName = packageName;
+        this.className = className;
+        this.classTemplate = classTemplate;
+        this.overwriteExisting = overwriteExisting;
+        this.relativePath = relativePath;
+    }
```

Comment:
This constructor is not necessary. An equivalent constructor would be generated by lombok via `@Value`. If you are getting errors relating to serialization when running tests in your IDE it is because you are missing the `-parameters` argument to the Java compiler.

Please also remove the similarly redundant constructors from the other classes.